infer.py

```diff
@@ -1,58 +1,78 @@
+		audio_name = os.path.basename(audio_path),
+		markup = transcripts.mask_to_intervals(speaker_mask, sample_rate),
+		sample_rate = sample_rate,
+		duration = signal.shape[-1] / sample_rate
+	)
+
 
 def main(args):
+	# load vad
+	if args.vad_type == 'simple':
+		_vad = vad.SimpleVAD(device=args.device)
+	elif args.vad_type == 'webrtc':
+		_vad = vad.WebrtcVAD(sample_rate=args.sample_rate)
+	else:
+		raise RuntimeError(f'VAD for type {args.vad_type} not found.')
+
 	# loading model
-	torch.set_grad_enabled(False)
-	model = models.SincTDNN(sample_rate = args.sample_rate, padding_same = True, tdnn = dict(kernel_size_pool = 501, stride_pool = 101))
-	print('stride:', model.stride / args.sample_rate, 'kernel_size:', model.kernel_size / args.sample_rate)
-	diag = model.load_state_dict(torch.load(args.weights_path), strict = False)
-	assert diag.missing_keys == ['sincnet_.conv1d_.0.window', 'sincnet_.conv1d_.0.sinct'] and diag.unexpected_keys == ['tdnn_.segment7.weight', 'tdnn_.segment7.bias']	
-	model.eval()
-	model.to(args.device)
-
-	# loading audio
-	# type: (shapes.BT, int) 
-	signal, sample_rate = audio.read_audio(args.audio_path, sample_rate = args.sample_rate, dtype = 'float32', mono = True, __array_wrap__ = torch.as_tensor)
-	print('read_audio done', signal.shape)
-	features : shapes.BCt = model(signal.to(args.device), args.sample_rate)
-	print('model done', features.shape)
-
-	# compute affinity
-	emb = features[0].t()
-	K = models.wang_affinity(emb)
-	W = (K > 0.5).float() # ???
-	L = models.normalized_symmetric_laplacian(W)
-	eigvals, eigvecs = L.symeig(eigenvectors = True)
-	num_eigvecs = 1
-	eigvecs = eigvecs[:, 1 : (1 + num_eigvecs)]
-	eigvecs_max = eigvecs.max()
-	fiedler_vector = eigvecs[:, 0] / eigvecs_max
-	print('fiedler_vector', fiedler_vector.shape)
-
-	silence = torch.zeros_like(fiedler_vector, dtype = torch.bool)
-	speaker1 = fiedler_vector < 0
-	speaker2 = fiedler_vector > 0
-
-	speaker_mask = torch.stack([silence, speaker1, speaker2])
-
-	# save transcript
-	transcript = transcripts.from_speaker_mask(speaker_mask, sample_rate)
-	print(transcripts.save(args.transcript_path, transcript))
-	
+	if args.model == 'spectral':
+		model = models.SpectralClusteringDiarizationModel(vad = _vad, vad_sensitivity = 0.90, weights_path = args.weights_path, device = args.device, sample_rate = args.sample_rate)
+	elif args.model == 'pyannote':
+		model = models.PyannoteDiarizationModel(vad = _vad, vad_sensitivity = 0.90, device = args.device, sample_rate = args.sample_rate)
+	else:
+		raise RuntimeError(f'Diarization model for name "{args.model}" not found.')
+
+	if args.audio_path[-5:] == '.json':
+		with open(args.audio_path) as dataset_file:
+			paths = [json.loads(line)['audio_path'] for line in dataset_file]
+	else:
+		paths = [args.audio_path]
+
+	# make transcripts
+	if args.workers > 0:
+		parametrized_transcript = functools.partial(get_transcript, model, sample_rate = args.sample_rate)
+		with mp.Pool(processes=args.workers) as pool:
```

Comment:
nit: please change to `--num-workers` and `num_workers` to simply match with convasr